src/reports.cpp

```diff
@@ -807,10 +807,12 @@ static int attack_info(const reports::context& rc, const attack_type &at, config
 		unsigned num_attacks = swarm_blows(min_attacks, max_attacks, cur_hp, max_hp);
 
 		color_t dmg_color = font::weapon_color;
-		if ( damage > specials_damage ) {
-			dmg_color = font::good_dmg_color;
-		} else if ( damage < specials_damage ) {
-			dmg_color = font::bad_dmg_color;
+		if(damage_multiplier != 100){
+			if ( damage > specials_damage ) {
+				dmg_color = font::good_dmg_color;
+			} else if ( damage < specials_damage ) {
+				dmg_color = font::bad_dmg_color;
+			}
 		}
```

Comment:
Haven't looked closely but this looks like it should be:
```c
if (damage > std::round(specials_damage)) {
    dmg_color = font::good_dmg_color;
} else if (damage < std::round(specials_damage)) {
    dmg_color = font::bad_dmg_color;
}
```

Reply:
@soliton-  fixed in https://github.com/wesnoth/wesnoth/pull/9476/commits/3003d223ee126b5f8f9274b86609cd45be83b6d1 commit